### ISSUE TITLE
K8s tests workflow is not reproducible on GitHub hosted runners

### ISSUE THREAD

**Issue by iteralube** [IndustryFusion/DigitalTwin#248]
I've tried to deploy the DigitalTwin Helm Chart using the instructions in the [README](https://github.com/iteralube/DigitalTwin/tree/main/helm#building-and-installation-of-platform-locally). I've tried this both locally in a VM and on GitHub hosted CI/CD runners via the K8s tests workflow included in the repository. The deployment fails on both platforms.

The problem is that the `minio-tenant` pod is not deployed successfully. Thus, the BATS test fails at this point. You can see the logs in this [job](https://github.com/iteralube/DigitalTwin/actions/runs/3418491400/jobs/5690851158), for example.

In the `minio-operator` pod, the following logs are present.
```shell
I1108 14:09:52.849051       1 main.go:70] Starting MinIO Operator
I1108 14:09:53.690661       1 main.go:169] caBundle on CRD updated
I1108 14:09:53.691052       1 main-controller.go:244] Setting up event handlers
I1108 14:09:53.691181       1 leaderelection.go:248] attempting to acquire leader lease minio-operator/minio-operator-lock...
I1108 14:09:53.796593       1 leaderelection.go:258] successfully acquired lease minio-operator/minio-operator-lock
I1108 14:09:53.796773       1 main-controller.go:478] minio-operator-7b7b79c4dd-9fn64: I am the leader, applying leader labels on myself
I1108 14:09:53.796885       1 main-controller.go:390] Waiting for API to start
I1108 14:09:53.796902       1 main-controller.go:382] Starting HTTP Upgrade Tenant Image server
I1108 14:09:53.952958       1 main-controller.go:351] Using Kubernetes CSR Version: v1
I1108 14:09:54.050419       1 operator.go:94] operator TLS secret not found: secrets "operator-tls" not found
I1108 14:09:54.112260       1 csr.go:183] Start polling for certificate of csr/operator-minio-operator-csr, every 5s, timeout after 20m0s
I1108 14:09:59.115402       1 csr.go:209] Certificate successfully fetched, creating secret with Private key and Certificate
I1108 14:09:59.118695       1 operator.go:96] Waiting for the operator certificates to be issued waiting for Operator cert
I1108 14:10:09.121772       1 main-controller.go:355] Starting HTTPS API server
I1108 14:10:09.121883       1 main-controller.go:393] Waiting for Upgrade Server to start
I1108 14:10:09.121896       1 main-controller.go:397] Starting Tenant controller
I1108 14:10:09.121901       1 main-controller.go:400] Waiting for informer caches to sync
I1108 14:10:09.121916       1 main-controller.go:405] Starting workers
I1108 14:11:01.366372       1 upgrades.go:91] Upgrading v4.2.0
I1108 14:11:01.368791       1 upgrades.go:111] iff-minio-tenant has no log secret
E1108 14:11:01.370949       1 upgrades.go:138] Error deleting operator webhook secret, manual deletion is needed: secrets "operator-webhook-secret" not found
I1108 14:11:01.379971       1 upgrades.go:91] Upgrading v4.2.4
I1108 14:11:01.384009       1 status.go:240] Hit conflict issue, getting latest version of tenant to update version
I1108 14:11:01.395399       1 upgrades.go:91] Upgrading v4.2.8
I1108 14:11:01.402820       1 status.go:240] Hit conflict issue, getting latest version of tenant to update version
I1108 14:11:01.410142       1 upgrades.go:91] Upgrading v4.2.9
I1108 14:11:01.414018       1 status.go:240] Hit conflict issue, getting latest version of tenant to update version
I1108 14:11:01.570066       1 upgrades.go:91] Upgrading v4.3.0
I1108 14:11:01.572390       1 upgrades.go:285] iff-minio-tenant has no log secret
I1108 14:11:01.768471       1 status.go:240] Hit conflict issue, getting latest version of tenant to update version
I1108 14:11:02.169666       1 upgrades.go:91] Upgrading v4.5
E1108 14:11:02.368756       1 main-controller.go:596] error syncing 'iff/iff-minio-tenant': error updating tenant 'iff/iff-minio-tenant', could not update tenant.spec.configuration field: Operation cannot be fulfilled on tenants.minio.min.io "iff-minio-tenant": the object has been modified; please apply your changes to the latest version and try again
I1108 14:11:11.382462       1 upgrades.go:91] Upgrading v4.5
E1108 14:11:11.395530       1 main-controller.go:596] error syncing 'iff/iff-minio-tenant': secrets "iff-minio-tenant-configuration" already exists
I1108 14:12:11.399164       1 upgrades.go:91] Upgrading v4.5
E1108 14:12:11.407523       1 main-controller.go:596] error syncing 'iff/iff-minio-tenant': secrets "iff-minio-tenant-configuration" already exists
I1108 14:13:09.132170       1 monitoring.go:100] 'iff/iff-minio-tenant' no pool is initialized
I1108 14:13:11.411066       1 upgrades.go:91] Upgrading v4.5
E1108 14:13:11.418176       1 main-controller.go:596] error syncing 'iff/iff-minio-tenant': secrets "iff-minio-tenant-configuration" already exists
```

I concluded that the secret `iff-minio-tenant-configuration` was the culprit, that prevented the successful deployment. Therefore, I deleted the secret via `kubectl`. This resulted in the following logs.
```shell
I1108 14:14:11.422474       1 upgrades.go:91] Upgrading v4.5
I1108 14:14:11.439652       1 status.go:240] Hit conflict issue, getting latest version of tenant to update version
E1108 14:14:11.448892       1 main-controller.go:596] error syncing 'iff/iff-minio-tenant': empty tenant credentials
I1108 14:15:11.473998       1 event.go:285] Event(v1.ObjectReference{Kind:"Tenant", Namespace:"iff", Name:"iff-minio-tenant", UID:"da576540-9a10-4f8d-9407-2c2d4200d008", APIVersion:"minio.min.io/v2", ResourceVersion:"3188", FieldPath:""}): type: 'Normal' reason: 'SvcCreated' MinIO Service Created
I1108 14:15:11.488822       1 status.go:54] Hit conflict issue, getting latest version of tenant
I1108 14:15:11.510430       1 event.go:285] Event(v1.ObjectReference{Kind:"Tenant", Namespace:"iff", Name:"iff-minio-tenant", UID:"da576540-9a10-4f8d-9407-2c2d4200d008", APIVersion:"minio.min.io/v2", ResourceVersion:"3194", FieldPath:""}): type: 'Normal' reason: 'SvcCreated' Console Service Created
I1108 14:15:11.522782       1 status.go:54] Hit conflict issue, getting latest version of tenant
I1108 14:15:11.543247       1 event.go:285] Event(v1.ObjectReference{Kind:"Tenant", Namespace:"iff", Name:"iff-minio-tenant", UID:"da576540-9a10-4f8d-9407-2c2d4200d008", APIVersion:"minio.min.io/v2", ResourceVersion:"3200", FieldPath:""}): type: 'Normal' reason: 'SvcCreated' Headless Service created
I1108 14:15:12.445933       1 minio.go:213] 'iff/operator-tls' secret not found, creating one now
E1108 14:15:12.651950       1 main-controller.go:596] error syncing 'iff/iff-minio-tenant': secrets "operator-tls" not found
I1108 14:16:09.132214       1 monitoring.go:100] 'iff/iff-minio-tenant' no pool is initialized
I1108 14:16:11.484710       1 main-controller.go:835] Detected we are updating a legacy tenant deployment
I1108 14:16:11.491728       1 main-controller.go:875] 'iff/iff-minio-tenant': Deploying pool ss-0
I1108 14:16:11.505359       1 event.go:285] Event(v1.ObjectReference{Kind:"Tenant", Namespace:"iff", Name:"iff-minio-tenant", UID:"da576540-9a10-4f8d-9407-2c2d4200d008", APIVersion:"minio.min.io/v2", ResourceVersion:"3293", FieldPath:""}): type: 'Normal' reason: 'PoolCreated' Tenant pool ss-0 created
I1108 14:16:26.680028       1 monitoring.go:130] 'iff/iff-minio-tenant' Failed to get cluster health: Get "http://minio.iff.svc.cluster.local/minio/health/cluster": dial tcp 10.43.101.246:80: i/o timeout
I1108 14:16:31.231429       1 http_handlers.go:180] MINIO_ARGS value is /export{0...3}
E1108 14:16:31.556413       1 main-controller.go:596] error syncing 'iff/iff-minio-tenant': context deadline exceeded
I1108 14:16:31.556578       1 event.go:285] Event(v1.ObjectReference{Kind:"Tenant", Namespace:"iff", Name:"iff-minio-tenant", UID:"da576540-9a10-4f8d-9407-2c2d4200d008", APIVersion:"minio.min.io/v2", ResourceVersion:"3298", FieldPath:""}): type: 'Warning' reason: 'UsersCreatedFailed' Users creation failed: context deadline exceeded
I1108 14:16:41.687740       1 monitoring.go:130] 'iff/iff-minio-tenant' Failed to get cluster health: Get "http://minio.iff.svc.cluster.local/minio/health/cluster": dial tcp 10.43.101.246:80: i/o timeout
I1108 14:17:12.227471       1 event.go:285] Event(v1.ObjectReference{Kind:"Tenant", Namespace:"iff", Name:"iff-minio-tenant", UID:"da576540-9a10-4f8d-9407-2c2d4200d008", APIVersion:"minio.min.io/v2", ResourceVersion:"3466", FieldPath:""}): type: 'Normal' reason: 'UsersCreated' Users created
I1108 14:17:12.288813       1 status.go:54] Hit conflict issue, getting latest version of tenant
```

After waiting a long enough time, the deployment completes successfully and all BATS tests pass.

I'm not sure, whether the deletion of the secrets has some unintended side effects, but as far as I could see it would just get recreated and would only have the metadata `uuid`, `creationTimestamp` and `resourceVersion` altered.

Am I doing something wrong or what is the best way to resolve this problem of the deployment?

**Comment by wagmarcel** [comment]
thanks for submitting the issue. I suspect that this secret was left-over from a previous unsuccessful or timeouted deployment. 
Question is why the redeployment did not work. Did you delete the whole helmfile deployment before redeploying?

**Comment by iteralube** [comment]
I tried the deployment on a local VM in VirtualBox. I saved the states between installation steps. As far as I am aware, there should be no previous deployment, if I restore to a previous saved state of the VM, where I didn't initiate the deployment yet. Additionally, on the hosted GitHub runners, there also shouldn't be any previous deployment, because as far as I am aware, the runners are cleaned before every run. Thus, I didn't delete the helmfile deployment explicitly, but also there shouldn't be any previous deployment in my scenario in the first place.
